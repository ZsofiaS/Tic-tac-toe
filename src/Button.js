import styled, { css } from 'styled-components';

const Button = styled.button`
  cursor: pointer;
  background: transparent;
  border-radius: 3px;
  color: palevioletred;
  border: 2px solid palevioletred;
  margin: 0 1em;
  padding: 0.25em 1em;
  transition: 0.5s all ease-out;

  // import css from styled components & interpolate a function into the template literal
  // the function gets passed the props of our component
  ${props =>
    props.primary &&
    css`
      color: crimson;
      border: 2px solid crimson;
    `};

  &:hover {
    background-color: ${(props) =>
      props.primary ? 'crimson' : 'palevioletred'};
    color: white;
  }
`;

export default Button;
