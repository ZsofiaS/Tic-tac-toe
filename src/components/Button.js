import styled, { css } from 'styled-components';

const Button = styled.button`
  cursor: pointer;
  background: transparent;
  border-radius: 3px;
  color: black;
  border: 2px solid;
  border-image-slice: 1;
  border-image-source: linear-gradient(to left, #118ab2, #06d6a0, #ffd166);
  background-image: linear-gradient(to left, #118ab2, #06d6a0, #ffd166);
  margin: 0 1em;
  padding: 0.5em 0.8em;
  transition: 0.3s all ease-out;
  font-family: 'Grandstander', cursive;
  font-size: 18px;
  color: white;

  // import css from styled components & interpolate a function into the template literal
  // the function gets passed the props of our component
  ${(props) =>
    props.primary &&
    css`
      color: crimson;
      border: 2px solid crimson;
    `};

  &:hover {
    background-color: none;
    background-image: none;
    color: black;
  }
`;

export default Button;
