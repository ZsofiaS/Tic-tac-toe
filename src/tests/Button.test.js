import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Button from '../components/Button';

describe('renders Button component', () => {
  it('renders without crashing', () => {
    shallow(<Button />);
  });
  it('should call onClick function when button clicked', () => {
    const mockedRestartGame = jest.fn();
    const restartGame = mockedRestartGame;
    const button = shallow(<Button onClick={restartGame} />);
    expect(mockedRestartGame).not.toHaveBeenCalled();
    button.simulate('click');
    expect(mockedRestartGame).toHaveBeenCalled();
  });
  it('renders Button correctly', () => {
    const buttonComponent = shallow(<Button />);
    expect(toJson(buttonComponent)).toMatchSnapshot();
  });
});
