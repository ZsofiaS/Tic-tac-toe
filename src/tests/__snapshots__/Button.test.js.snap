// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Button component renders Button correctly 1`] = `
<button
  className="sc-bdnylx bbDyYP"
/>
`;
